sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiDynamicDestinationsBeamRow.java

```diff
@@ -61,9 +61,15 @@ public TableSchema getTableSchema() {
     }
 
     @Override
+    @SuppressWarnings("nullness")
     public StorageApiWritePayload toMessage(T element) {
       Message msg = BeamRowToStorageApiProto.messageFromBeamRow(descriptor, toRow.apply(element));
-      return new AutoValue_StorageApiWritePayload(msg.toByteArray());
+      return new AutoValue_StorageApiWritePayload(msg.toByteArray(), null);
+    }
+
+    @Override
+    public StorageApiWritePayload toMessage(TableRow tableRow, boolean respectRequired) {
+      throw new RuntimeException("Not supported");
```

Comment:
Maybe I misunderstood the logic, but would it be possible to add a validation in BigQueryIO that schema update is not available for Beam Row payloads? 
If I'm not mistaken someone could configure the writes with `useBeamSchema()` and set also auto update of schemas and then their pipeline will fail with this runtime exception, when in fact we could have captured that in validation time.

Reply:
done